include/matrix.h

```diff
@@ -184,13 +186,58 @@ bool operator!=(Matrix<TDataType, TSize1, TSize2> const& First,
 template <typename TDataType, std::size_t TSize1, std::size_t TSize2>
 inline std::ostream& operator<<(std::ostream& rOStream,
     Matrix<TDataType, TSize1, TSize2> const& TheMatrix) {
-    rOStream << '{';
+    auto matrixPrefix = "[";
+    auto matrixSuffix = "]";
```

Comment:
I would use `constexpr char` for the single character ones to improve the performance.

Reply:
In case we want to choose between different formats I would suggest something like this:
```cpp
struct Format
{
    std::string matrix_prefix = "[";
    std::string matrix_suffix = "]";
    std::string row_prefix = "[";
    std::string row_suffix = "]";
    std::string row_separator = ",\n ";
    std::string col_separator = ", ";
    ...
};

const Format JsonFormat = { "[", "]", "[", "]", ",\n ", ", " };
const Format MathematicaFormat = { "{", "}", "{", "}", ", ", ", " };
```

And make it useable like this:
```cpp
std::cout << JsonFormat(a_matrix) << std::endl;
std::cout << MathematicaFormat(a_matrix) << std::endl;
```